src/NITTA/Model/Problems/Allocation.hs

```diff
@@ -0,0 +1,39 @@
+{-# LANGUAGE ConstraintKinds #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE FunctionalDependencies #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+{- |
+Module      : NITTA.Model.Problems.Allocation
+Description : Function distribution between processor units
+Copyright   : (c) Aleksandr Penskoi, 2022
+License     : BSD3
+Maintainer  : aleksandr.penskoi@gmail.com
+Stability   : experimental
+-}
+module NITTA.Model.Problems.Allocation (
+    Allocation (..),
+    AllocationProblem (..),
+) where
+
+import Data.String.ToString (ToString (..))
+import GHC.Generics (Generic)
+
+data Allocation tag = Allocation
```

Comment:
UnitPrototype depends on `PUClasses` (NITTA.Model.Networks.Types) and NITTA.Model.Networks.Types module depend on NITTA.Model.Problems. So I can't place UnitPrototype here because of cycled dependencies